pkg/resources/kafka/configmap.go

```diff
@@ -250,3 +255,116 @@ func (r Reconciler) generateBrokerConfig(id int32, brokerConfig *v1beta1.BrokerC
 
 	return strings.Join(completeConfig, "\n")
 }
+
+func (r *Reconciler) postConfigMapReconcile(log logr.Logger, desired *corev1.ConfigMap) error {
+	desiredType := reflect.TypeOf(desired)
+	current := desired.DeepCopyObject()
+
+	key, err := runtimeClient.ObjectKeyFromObject(current)
+	if err != nil {
+		return errors.WithDetails(err, "kind", desiredType)
+	}
+	log = log.WithValues("kind", desiredType, "name", key.Name)
+
+	err = r.Client.Get(context.TODO(), key, current)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errorfactory.New(errorfactory.APIFailure{}, err, "getting resource failed", "kind", desiredType, "name", key.Name)
+	}
+
+	if k8sutil.CheckIfObjectUpdated(log, desiredType, current, desired) {
+		// Only update status when configmap belongs to broker
+		if id, ok := desired.Labels["brokerId"]; ok {
+			currentConfigs := getBrokerConfigsFromConfigMap(current.(*corev1.ConfigMap))
+			desiredConfigs := getBrokerConfigsFromConfigMap(desired)
+
+			var statusErr error
+			// if only per broker configs are changed, do not trigger rolling upgrade by setting ConfigOutOfSync status
+			if shouldRefreshOnlyPerBrokerConfigs(currentConfigs, desiredConfigs, log) {
+				log.V(1).Info("setting per broker config status to out of sync")
+				statusErr = k8sutil.UpdateBrokerStatus(r.Client, []string{id}, r.KafkaCluster, v1beta1.PerBrokerConfigOutOfSync, log)
+			} else {
+				statusErr = k8sutil.UpdateBrokerStatus(r.Client, []string{id}, r.KafkaCluster, v1beta1.ConfigOutOfSync, log)
+			}
+			if statusErr != nil {
+				return errors.WrapIfWithDetails(err, "updating status for resource failed", "kind", desiredType)
+			}
+		}
+	}
+
+	return nil
+}
+
+func shouldRefreshOnlyPerBrokerConfigs(currentConfigs, desiredConfigs map[string]string, log logr.Logger) bool {
+	touchedConfigs := collectTouchedConfigs(currentConfigs, desiredConfigs, log)
+
+OUTERLOOP:
+	for _, config := range touchedConfigs {
```

Comment:
This part could be refactored as follows to make it more readable:
```
touchedConfigs := collectTouchedConfigs(currentConfigs, desiredConfigs, log)

for _, perBrokerConfig := range perBrokerConfigs {
  for _, config := range touchedConfigs {
           if config != perBrokerConfig {
              return false // broker config that can not be updated dynamicaaly has been modified
           }

 		   if listenersSecurityProtocolChanged(...) {
                return false
           }
  }
}

return true
```

Reply:
Actually if we return with a set data structure from `collectTouchedConfigs` we can do this much simpler. I come up with the following solution:
```go
	touchedConfigs := collectTouchedConfigs(currentConfigs, desiredConfigs, log)
	if _, ok := touchedConfigs[securityProtocolMapConfigName]; ok {
		if listenersSecurityProtocolChanged(currentConfigs, desiredConfigs) {
			return false
		}
	}
	for _, perBrokerConfig := range perBrokerConfigs {
		delete(touchedConfigs, perBrokerConfig)
	}
	return len(touchedConfigs) == 0
```
Assuming `collectTouchedConfigs` returns with `map[string]struct{}`